primitives/src/vstaging/mod.rs

```diff
@@ -30,6 +30,58 @@ use parity_util_mem::MallocSizeOf;
 /// Useful type alias for Para IDs.
 pub type ParaId = Id;
 
+/// Candidate's acceptance limitations for asynchronous backing per relay parent.
+#[derive(RuntimeDebug, Copy, Clone, PartialEq, Encode, Decode, TypeInfo)]
+#[cfg_attr(feature = "std", derive(MallocSizeOf, serde::Serialize, serde::Deserialize))]
+pub struct AsyncBackingParameters {
+	/// The maximum number of para blocks between the para head in a relay parent
+	/// and a new candidate. Restricts nodes from building arbitrary long chains
+	/// and spamming other validators.
+	///
+	/// When async backing is disabled, the only valid value is 0.
+	pub max_candidate_depth: u32,
+	/// How many ancestors of a relay parent are allowed to build candidates on top
+	/// of.
+	///
+	/// When async backing is disabled, the only valid value is 0.
+	pub allowed_ancestry_len: u32,
+}
+
+/// Abridged version of `HostConfiguration` (from the `Configuration` parachains host runtime module)
+/// meant to be used by a parachain or PDK such as cumulus.
+#[derive(Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+#[cfg_attr(feature = "std", derive(PartialEq))]
+pub struct AbridgedHostConfiguration {
+	/// Asynchronous backing parameters.
+	pub async_backing_params: AsyncBackingParameters,
```

Comment:
are you foreseeing a particular usecase where this is needed on the Cumulus side?

Reply:
In parachains-system there's a [check](https://github.com/paritytech/cumulus/blob/26e17fea7ef95d91b94db92079034d4355ad3658/pallets/parachain-system/src/lib.rs#L359-L364) performed on a last relay chain block number.
It ensures that the number strictly increases.

Most importantly, a para cannot progress more than 1 block per relay chain block. And this could be lifted with a `max_candidate_depth`. In fact, we need to check that there're no more than `max_candidate_depth` non-included candidates in the allowed ancestry, which is more complicated.

Less importantly, nothing (in theory) restricts a para from basing the next candidate on an older relay parent than the previous one. I don't see how this is someone's usecase but anyway.